build-config/make/images.make

```diff
@@ -387,20 +418,36 @@ recovery-iso: $(RECOVERY_ISO_STAMP)
 $(RECOVERY_ISO_STAMP): $(GRUB_HOST_INSTALL_STAMP) $(RECOVERY_INITRD_STAMP) \
 			$(RECOVERY_CONF_DIR)/grub-iso.cfg $(RECOVERY_XORRISO_OPTIONS)
 	$(Q) echo "==== Create $(MACHINE_PREFIX) ONIE Recovery Hybrid iso ===="
-	$(Q) Q=$(Q) CONSOLE_SPEED=$(CONSOLE_SPEED) \
+    ifeq ($(ARCH),arm64)
```

Comment:
Like the above I think this all would collapse down if you rename the original `onie-mk-iso.sh` to `onie-mk-iso-x86_64.sh`.  Then this file can refer to `onie-mk-iso-$(ARCH).sh`.

I haven't read the patches yet, but I do not understand why a separate `onie-mk-iso.sh` is necessary at all.

If the differences are minor another suggestion is to only have one script, `onie-mk-iso.sh`, but pass in the $(ARCH) as a parameter and take appropriate action within the script.